rllib/examples/rl_modules/classes/lstm_containing_rlm.py

```diff
@@ -106,70 +106,50 @@ def get_initial_state(self) -> Any:
         }
 
     @override(TorchRLModule)
-    def _forward_inference(self, batch, **kwargs):
+    def _forward(self, batch, **kwargs):
         # Compute the basic 1D feature tensor (inputs to policy- and value-heads).
-        _, state_out, logits = self._compute_features_state_out_and_logits(batch)
+        features, state_outs = self._compute_features_and_state_outs(batch)
+        logits = self._pi_head(features)
 
         # Return logits as ACTION_DIST_INPUTS (categorical distribution).
         # Note that the default `GetActions` connector piece (in the EnvRunner) will
         # take care of argmax-"sampling" from the logits to yield the inference (greedy)
         # action.
         return {
-            Columns.STATE_OUT: state_out,
             Columns.ACTION_DIST_INPUTS: logits,
+            Columns.STATE_OUT: state_outs,
         }
 
-    @override(TorchRLModule)
-    def _forward_exploration(self, batch, **kwargs):
-        # Exact same as `_forward_inference`.
-        # Note that the default `GetActions` connector piece (in the EnvRunner) will
-        # take care of stochastic sampling from the Categorical defined by the logits
-        # to yield the exploration action.
-        return self._forward_inference(batch, **kwargs)
-
     @override(TorchRLModule)
     def _forward_train(self, batch, **kwargs):
-        # Compute the basic 1D feature tensor (inputs to policy- and value-heads).
-        features, state_out, logits = self._compute_features_state_out_and_logits(batch)
-        # Besides the action logits, we also have to return value predictions here
-        # (to be used inside the loss function).
-        values = self._values(features).squeeze(-1)
+        # Same logic as _forward, but also return features to be used by value function
+        # branch during training.
+        features, state_outs = self._compute_features_and_state_outs(batch)
```

Comment:
As before, in my very own opinion I think "features" is a misleading name as it is usually used for the inputs of a neural network.

Reply:
Fixed everywhere. Great catch and suggestion! Makes things much clearer.